internal/provider/config_resource.go

```diff
@@ -421,5 +475,21 @@ func transformToResourceModel(d *client.MTEConfigDto) MTEConfigModel {
 		}
 	}
 
+	if len(d.ConditionalHeaders) != 0 {
+
```

Comment:
This logic appears to be some duplication of logic on line 410, can we perhaps extract this to a function?

Reply:
have refactored to match the architecture of the other fields in the resource and api models, see, what you think of that